addons/product/static/src/js/pricelist_report/product_pricelist_report.js

```diff
@@ -183,6 +217,70 @@ export class ProductPricelistReport extends Component {
         });
     }
 
+    async export_csv() {
+        try {
+            const response = await this.orm.call('report.product.report_pricelist', 'export_csv', [], { data: this.reportParams });
+            const { csv_data, file_name } = response;
+
+            const csvContent = `data:text/csv;charset=utf-8,${csv_data}`;
+            const encodedUri = encodeURI(csvContent);
+
+            const link = document.createElement('a');
+            link.href = encodedUri;
+            link.download = file_name;
+            document.body.appendChild(link);
+            link.click();
+            document.body.removeChild(link);
+        } catch (error) {
+            console.error('Error exporting CSV:', error);
+            await this.action.doAction(
+                sendCustomNotification('danger', _t("Error exporting CSV file. Please try again."))
+            );
+        }
```

Comment:
@jeep-odoo did you try using the `download` utils from the framework (with dedicated routes instead of an orm call ?)
I believe it could simplify the logic here.

Some examples can be found for xls and csv in `spreadsheet_download_action.js` & `generate_warrant_payslips.js`